### DISCUSSION TITLE
How does system handle multiple users wanting use different model simultaneously?

### DISCUSSION THREAD

**Discussion by Shawneau** [open-webui/open-webui#816] [Q&A] [answered]
Is that even possible? Is it FIFO?

Should I just have one model available if I'm hosting multiple users?

Thanks!

**Comment by justinh-rahb** [comment, marked as answer]
Ollama is essentially FIFO, yes. Requests are blocking and queued. The WebUI will only wait for so long so this may result in an error. One way around this is harnessing a herd of Ollamas together with a load-balancer but this obviously requires either multiple machines or a machine with multiple GPUs. An M1/M2/M3 Max with a financially irresponsible amount of RAM can run 2 instances, though things get a bit slower when they're all fighting for the CPU and GPU at the same time. In practice, I've got about a dozen users sharing a 96GB M2 Max running 2 Ollamas behind the WebUI and we rarely get multiple concurrent requests anyway, but on the chance we do it handles it fine.

**Reply by ctxadm** [reply]
very usefull, thank :)

**Reply by UberMetroid** [reply]
> load-balancer

Any recommendations?
